protocols/identify/src/handler.rs

```diff
@@ -207,14 +299,37 @@ impl ConnectionHandler for Handler {
         SubstreamProtocol::new(SelectUpgrade::new(Protocol, PushProtocol::inbound()), ())
     }
 
-    fn on_behaviour_event(&mut self, Push(push): Self::InEvent) {
-        self.events
-            .push(ConnectionHandlerEvent::OutboundSubstreamRequest {
-                protocol: SubstreamProtocol::new(
-                    EitherUpgrade::B(PushProtocol::outbound(push)),
-                    (),
-                ),
-            });
+    fn on_behaviour_event(&mut self, event: Self::InEvent) {
+        match event {
+            InEvent::Push(push) => {
+                self.events
+                    .push(ConnectionHandlerEvent::OutboundSubstreamRequest {
+                        protocol: SubstreamProtocol::new(
+                            EitherUpgrade::B(PushProtocol::outbound(push)),
+                            (),
+                        ),
+                    });
+            }
+            InEvent::Identify(behaviour_info) => {
+                let info = Info {
+                    public_key: self.public_key.clone(),
+                    protocol_version: self.protocol_version.clone(),
+                    agent_version: self.agent_version.clone(),
+                    listen_addrs: behaviour_info.listen_addrs,
+                    protocols: behaviour_info.protocols,
+                    observed_addr: self.observed_addr.clone(),
+                };
+                let substream = self
+                    .reply_streams
+                    .pop_front()
+                    .expect("A BehaviourInfo reply should have a matching substream.");
+                let io = Box::pin(substream.send(info));
```

Comment:
```suggestion
                let io = Box::pin(async move { substream.send(info).await?; Ok(self.remote_peer_id) });
```

If you construct the future this way, you don't need the `Sending` struct and can you `FuturesUnordered` instead :)

Reply:
ah! How nice, thanks Thomas! :)